src/Symfony/Component/HttpFoundation/ChainRequestMatcher.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+/**
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class ChainRequestMatcher implements RequestMatcherInterface
```

Comment:
Is this class related to this PR? It's not used anywhere?
About the name, "chain" usually means "delegation chain", ie the first applicable in the chain takes responsibility for the job. Here, this is something else, like `AllMatchersRequestMatcher` (dummy suggestion, I'm not really proud of it :) )